### PR TITLE
Remove API Client from the tree

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

Branch to parallel my cleanup of the API Client in order for it to be removed from this repository.

The API Client side is at https://github.com/jellyfin/jellyfin-apiclient-javascript/pull/36

**Issues**

### CODE REVIEW

**Comment by dkanada** [comment]
What a beautiful reduction in code :smile: can't wait to get this through.

**Comment by ThibaultNocchi** [comment]
Maybe you'll be able to help as I'm having problem with a "ReferenceError: ApiClient is not defined" error in controllers/wizard/start.js.

I'm using the latest server commit and replacing its web folder with the latest web client commit. I'm being normally redirected to the wizard and I'm hit with this error. And trying to get `window.ApiClient` from the JS console returns me an undefined value.

**Comment by MrTimscampi** [comment]
@jellyfin/web This depends on https://github.com/jellyfin/jellyfin-apiclient-javascript/pull/38, which is under final cleanup.

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "hls.js": "^0.13.1",
     "howler": "^2.1.3",
     "intersection-observer": "^0.7.0",
+    "jellyfin-apiclient": "https://github.com/MrTimscampi/jellyfin-apiclient-javascript#jf-web-part2",
```
Comment (ferferga) [inline package.json:68]: I guess we want to change that to our repo?

File: yarn.lock [hunk 2]
```diff
@@ -6542,6 +6542,10 @@ isurl@^1.0.0-alpha5:
     has-to-string-tag-x "^1.2.0"
     is-object "^1.0.1"
 
+"jellyfin-apiclient@https://github.com/MrTimscampi/jellyfin-apiclient-javascript#jf-web-part2":
```
Comment (ferferga) [inline yarn.lock:6545]: Same here

File: package.json (hunk 1 above)
Comment (MrTimscampi) [reply package.json:68]: This is in progress, I'm waiting on a tag for 1.1.0 to pin this against the org's repository.

File: yarn.lock (hunk 2 above)
Comment (MrTimscampi) [reply yarn.lock:6545]: This will be changed automatically once the one in package.json is taken care of.

File: package.json [hunk 3]
```diff
@@ -90,18 +91,19 @@
         "test": [
           "src/components/autoFocuser.js",
           "src/components/cardbuilder/cardBuilder.js",
-          "src/scripts/dom.js",
           "src/components/filedownloader.js",
           "src/scripts/filesystem.js",
           "src/scripts/keyboardnavigation.js",
```
Comment (dmitrylyzo) [inline package.json:96]: dup
```suggestion
```

File: src/bundle.js [hunk 4]
```diff
@@ -152,3 +153,26 @@ var headroom = require("headroom.js/dist/headroom");
...
+    return apiclient.Credentials;
+});
+
+_define("connectionManagerFactory", function () {
+    return apiclient.ConnectionManager;
+});
+
+_define("appStorage", function () {
+    return apiclient.AppStorage;
+});
```
Comment (JustAMan) [inline src/bundle.js:178]: why are those needed, why cannot we just use `apiclient.Events` instead of `events`?

File: webpack.dev.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ module.exports = merge(common, {
         path: path.resolve(__dirname, "dist"),
         libraryTarget: "amd-require"
     },
-    devtool: '#inline-source-map',
+    devtool: 'inline-source-map',
```
Comment (JustAMan) [inline webpack.dev.js:12]: can you explain?

File: src/bundle.js (hunk 4 above)
Comment (MrTimscampi) [reply src/bundle.js:178]: It's (for now) to do as few code modifications as needed.

File: webpack.dev.js (hunk 5 above)
Comment (MrTimscampi) [reply webpack.dev.js:12]: The docs doesn't have the `#`.

File: .eslintrc.js [hunk 6]
```diff
@@ -42,7 +42,9 @@ module.exports = {
         'one-var': ["error", "never"],
         'quotes': ["error", "single", { "avoidEscape": true, "allowTemplateLiterals": false }],
         'semi': ["error"],
-        'space-before-blocks': ["error"]
+        'space-before-blocks': ["error"],
+        // Disable while it's buggy
+        'compat/compat': ["off"]
```
Comment (JustAMan) [inline .eslintrc.js:47]: cannot we shut it up only in a place we know it's buggy? not blanket everywhere?

File: .ci/azure-pipelines.yml [hunk 7]
```diff
@@ -87,16 +87,16 @@ jobs:
     inputs:
       versionSpec: '12.x'
 
-  - task: Cache@2
-    displayName: 'Check Cache'
-    inputs:
-      key: 'yarn | yarn.lock'
-      path: 'node_modules'
-      cacheHitVar: CACHE_RESTORED
+  #- task: Cache@2
```
Comment (JustAMan) [inline .ci/azure-pipelines.yml]: should probably return caching as apiclient is now being get from npm store

File: package.json [hunk 8]
```diff
@@ -92,13 +92,14 @@
...
           "src/components/filedownloader.js",
-          "src/components/playback/mediasession.js",
           "src/components/images/imageLoader.js",
           "src/components/lazyloader/lazyloader-intersectionobserver.js",
+          "src/components/playback/mediasession.js",
           "src/components/sanatizefilename.js",
           "src/components/scrollManager.js",
           "src/scripts/dfnshelper.js",
           "src/scripts/dom.js",
+          "src/scripts/filesystem.js",
```
Comment (JustAMan) [inline package.json]: why add?.. bad merge?

File: src/scripts/site.js [hunk 9]
```diff
@@ -355,7 +355,7 @@ var AppInfo = {};
     }
 
     function onRequireJsError(requireType, requireModules) {
-        throw new Error("error loading " + requireModules);
+        console.error('RequireJS error: ' + (requireType || 'unknown') + '. Failed modules: ' + (requireModules || []).join(','));
```
Comment (JustAMan) [inline src/scripts/site.js]: why?

File: src/scripts/site.js [hunk 10]
```diff
@@ -398,11 +396,12 @@ var AppInfo = {};
 
         define('prompt', [componentsPath + '/prompt/prompt'], returnFirstDependency);
 
-        define("loading", [componentsPath + "/loading/loading"], returnFirstDependency);
-        define("multi-download", [componentsPath + "/multidownload"], returnFirstDependency);
-        define("fileDownloader", [componentsPath + "/filedownloader"], returnFirstDependency);
+        define('loading', [componentsPath + '/loading/loading'], returnFirstDependency);
+        define('multi-download', [componentsPath + '/multidownload'], returnFirstDependency);
+        define('fileDownloader', [componentsPath + '/filedownloader'], returnFirstDependency);
+        define('localassetmanager', [bowerPath + '/apiclient/localassetmanager'], returnFirstDependency);
```
Comment (JustAMan) [inline src/scripts/site.js:402]: why new `define`?

File: package.json (hunk 8 above)
Comment (JustAMan) [reply package.json]: turns out it's bad diff

File: src/scripts/site.js (hunk 9 above)
Comment (JustAMan) [reply src/scripts/site.js]: bad diff

File: src/scripts/site.js [hunk 11]
```diff
@@ -404,10 +402,6 @@ var AppInfo = {};
         define('localassetmanager', [bowerPath + '/apiclient/localassetmanager'], returnFirstDependency);
```
Comment (JustAMan) [inline src/scripts/site.js:402]: why this was not changed?

File: src/scripts/site.js (hunk 10 above)
Comment (JustAMan) [reply src/scripts/site.js:402]: bad diff

File: src/scripts/site.js (hunk 11 above)
Comment (MrTimscampi) [reply src/scripts/site.js]: Oversight when merging master, fixed in latest commit

File: .eslintrc.js (hunk 6 above)
Comment (MrTimscampi) [reply .eslintrc.js]: I changed the way it works by adding the offending API to the polyfills temporarily. Had to disable one more line which wasn't caught before, for some reason.
